src/App.js

```diff
@@ -73,6 +74,73 @@ function App() {
     setSearchText("");
   };
 
+  //filter state
+
+  const [withoutOxygen, setWithoutOxygen] = React.useState(false);
+  const [oxygen, setOxygen] = React.useState(false);
+  const [icu, setIcu] = React.useState(false);
+  const [icuwithventilator, setIcuWithVentilator] = React.useState(false);
+
+  const handleIcu = () => {
+    setIcu(!icu);
+    setOxygen(false);
+    setWithoutOxygen(false);
+    setIcuWithVentilator(false);
+  };
+
+  const handleOxygen = () => {
+    setOxygen(!oxygen);
+    setIcu(false);
+    setWithoutOxygen(false);
+    setIcuWithVentilator(false);
+  };
+
+  const handleIcuWithVentilator = () => {
+    setIcuWithVentilator(!icuwithventilator);
+    setOxygen(false);
+    setWithoutOxygen(false);
+    setIcu(false);
+  };
+
+  const handleWithoutOxygen = () => {
+    setWithoutOxygen(!withoutOxygen);
+    setOxygen(false);
+    setIcu(false);
+    setIcuWithVentilator(false);
+  };
+
+  useEffect(() => {
+    if (oxygen) {
+      const oxygenFilteredArr = hospitalList[selectedDist].filter(
+        (hospital) => {
+          return hospital["OXYGEN BEDS Vacant"] > 0;
+        }
+      );
+      setFilteredHospitalList(oxygenFilteredArr);
+    } else if (icu) {
+      const icufilteredArr = hospitalList[selectedDist].filter((hospital) => {
+        return hospital["ICU BEDS Vacant"] > 0;
+      });
+      setFilteredHospitalList(icufilteredArr);
+    } else if (withoutOxygen) {
+      const withoutOxygenFilteredArr = hospitalList[selectedDist].filter(
+        (hospital) => {
+          return hospital["NON-OXYGEN SUPPORTED BEDS Vacant"] > 0;
+        }
+      );
+      setFilteredHospitalList(withoutOxygenFilteredArr);
+    } else if (icuwithventilator) {
+      const icuWithVentilatorFilteredArr = hospitalList[selectedDist].filter(
+        (hospital) => {
+          return hospital["VENTILATOR Vacant"] > 0;
+        }
+      );
+      setFilteredHospitalList(icuWithVentilatorFilteredArr);
+    } else {
+      setFilteredHospitalList(hospitalList[selectedDist]);
+    }
+  }, [icu, oxygen, withoutOxygen, icuwithventilator, selectedDist]);
```

Comment:
also here, just pass in the `setFilterState` and `selectedDist` for watch. and the if-else can be simplified to a single `filter`. by using some kind of global object. like

```
const badges = {
	oxygen: {
		prop: "OXYGEN BEDS Vacant"
		label: "With Oxygen"
		action: () => /*Not sure if you will need an action callback. I will leave to you here.*/
	}
	...
	...
}
```